File: toyORBJava/src/MessageMarshaller/Marshaller.java
package MessageMarshaller;

public class Marshaller
{
	static public byte[] marshal(Message theMsg)
	{
		String m = "  " + theMsg.sender + ":" + theMsg.data;
		byte b[] = new byte[m.length()];
		b = m.getBytes();
		b[0] = (byte)m.length();
		return b;
	}
	static public Message unmarshal(byte[] byteArray)
	{
		String msg = new String(byteArray);
		String sender = msg.substring(1, msg.indexOf(":"));
		System.out.println("Sender:" + sender);
		String m = msg.substring(msg.indexOf(":")+1, msg.length()-1);
		return new Message(sender, m);
	}

}

File: ToyORBC#/ToyORB/Marshaller.cs
using System.Net.Mime;
using System.Text;
using System.Xml;

namespace ToyORB;

public class Marshaller
{
    static public byte[] marshal(Message theMsg)
    {
        string m = "  " + theMsg.sender + ":" + theMsg.data;
        var b = Encoding.ASCII.GetBytes(m);
        b[0] = (byte)m.Length;
        return b;
    }
    static public Message unmarshal(byte[] byteArray)
    {
        Console.WriteLine();
        string receivedText = Encoding.UTF8.GetString(byteArray, 0, byteArray[0]);
        string sender = receivedText.Substring(1, receivedText.IndexOf(":")).Trim();
        string data = receivedText.Substring(receivedText.IndexOf(':')+1).Trim();
        return new Message(sender, data);
    }
}